identity/ideth/service.go

```diff
@@ -347,12 +348,37 @@ func (i service) ValidateKey(ctx context.Context, did id.DID, key []byte, purpos
 		return err
 	}
 
-	keys, err := contract.GetKey(opts, key32)
+	ethKey, err := contract.GetKey(opts, key32)
 	if err != nil {
 		return err
 	}
 
-	for _, p := range keys.Purposes {
+	// if revoked
+	if ethKey.RevokedAt.Cmp(big.NewInt(0)) > 0 {
+		// if a specific time for validation is provided then we validate if a revoked key was revoked before the provided time
+		if validateAt != nil {
+			revokedAtBlock, err := i.client.GetEthClient().BlockByNumber(ctx, ethKey.RevokedAt)
+			if err != nil {
+				return err
+			}
+
+			if big.NewInt(validateAt.Unix()).Cmp(revokedAtBlock.Time()) > 0 {
+				return errors.New("the given key [%x] for purpose [%s] has been revoked before provided time %s", key, purpose.String(), validateAt.String())
+			}
+		} else {
+			// else check if the revocation block number is before the latest block number from Ethereum
+			latest, err := i.client.GetEthClient().BlockByNumber(ctx, nil)
```

Comment:
I'm not sure if this is the correct check. If Alice signs a document at block 3 and key was revoked at 2 and latest block is at 6, this would still go through right ?

Shouldn't we be checking the blcok at which the document was anchored ?

Reply:
its a fall back, the actual check we should use is the one with the time.

Reply:
I will remove this else block